bot/exts/info/doc/_cog.py

```diff
@@ -248,7 +252,10 @@ async def get_symbol_embed(self, symbol: str) -> Optional[discord.Embed]:
             return None
         self.bot.stats.incr(f"doc_fetches.{symbol_info.package}")
 
+        self.symbol_get_event.clear()
         markdown = await doc_cache.get(symbol_info)
+        self.symbol_get_event.set()
```

Comment:
I've realized that this won't help if there are multiple coros suspended here, as one can set the event while the other is still waiting leading to the issue this tires to prevent.
Does something exist in the stdlib that'd allow us to count the event clears and only let the refresh to take control when every coro calls set or an equivalent?

Reply:
If I understand correctly, here's what we want to happen:

* If an inventory is being refreshed, make all `get_symbol_embed` calls wait.
* Allow many concurrent calls of `get_symbol_embed`.
* Make an inventory refresh wait for all calls of `get_symbol_embed` to complete.

I don't think anything exists that can directly solve this exact problem.